gluten-core/src/main/scala/io/glutenproject/execution/ExpandExecTransformer.scala

```diff
@@ -84,160 +81,148 @@ case class ExpandExecTransformer(projections: Seq[Seq[Expression]],
 
   def getRelNode(context: SubstraitContext,
                  projections: Seq[Seq[Expression]],
-                 groupExpression: Seq[NamedExpression],
                  originalInputAttributes: Seq[Attribute],
                  operatorId: Long,
                  input: RelNode,
                  validation: Boolean): RelNode = {
     val args = context.registeredFunction
-    val groupSize = groupExpression.size
-    val aggSize = projections.head.size - groupSize
-
-    val groupsetExprNodes = new util.ArrayList[util.ArrayList[ExpressionNode]]()
-    val aggExprNodes = new util.ArrayList[ExpressionNode]()
-    for (i <- 0 until aggSize) {
-      val aggExprNode = ExpressionConverter
-        .replaceWithExpressionTransformer(
-          projections.head(i),
-          originalInputAttributes)
-        .doTransform(args)
-      aggExprNodes.add(aggExprNode)
+    def needsPreProjection(projections: Seq[Seq[Expression]]): Boolean = {
+      projections
+        .exists(set => set.exists(p => !p.isInstanceOf[Attribute] && !p.isInstanceOf[Literal]))
     }
+    if (needsPreProjection(projections)) {
```

Comment:
Can we add simple unit test for this case?

Reply:
Added, thanks.
![image](https://user-images.githubusercontent.com/10524738/232685654-3385ec48-44cb-4f80-9267-d886d0e0595b.png)